tanjun/components.py

```diff
@@ -715,9 +717,25 @@ def remove_client_callback(self, name: str, callback: tanjun.MetaEventSig, /) ->
         if self._client:
             self._client.remove_client_callback(name, callback)
 
+    @typing.overload
+    def with_client_callback(
```

Comment:
The `as_` load_from_scope functions shouldn't replace these decorators, they should just be added alongside this (and in the case of event listeners just directly call the with_listener instead of add_listener to allow for type hint analysis)